lib/smb/smb/zgrab.go

```diff
@@ -269,6 +279,62 @@ func (ls *LoggedSession) LoggedNegotiateProtocolv1(setup bool) error {
 	return nil
 }
 
+func (ls *LoggedSession) LoggedSessionSetupV1() (err error) {
+	s := &ls.Session
+	var buf []byte
+
+	req := s.NewSessionSetupV1Req()
+	s.Debug("Sending LoggedSessionSetupV1 Request", nil)
+	buf, err = s.send(req)
+	if err != nil {
+		s.Debug("No response to SMBv1 cleartext SessionSetup", nil)
+		return nil
+	}
+
+	// Safely trim down everything except the payload
+	if len(buf) < SmbHeaderV1Length {
+		return nil
+	}
+	// When using unicode, a padding byte will exist after the header
+	paddingLength := int((buf[11] >> 7) & 1)
+	// Skip header
+	buf = buf[SmbHeaderV1Length:]
+	// The byte after the header holds the number of words in uint16s
+	if len(buf) < (int(buf[0])*2)+3+paddingLength {
+		return nil
+	}
+	// words + 3 bytes for wordlength & bytecount + potential unicode padding
+	buf = buf[(int(buf[0])*2)+3+paddingLength:]
```

Comment:
This index value (`(int(buf[0])*2)+3+paddingLength:`) is validated above, but I recommend moving this out to a variable.  That will make it easier to verify that the validation is happening against the same value.

Reply:
Good call